appinventor/components/src/com/google/appinventor/components/runtime/Clock.java

```diff
@@ -207,6 +216,102 @@ public static Calendar MakeInstant(String from) {
     }
   }
 
+  /**
+   * An instant in time specified by MM/DD/YYYY 
+   * where MM is the month (01-12), DD the day (01-31), YYYY the year
+   * (0000-9999)
+   * @param year year integer
+   * @param month month integer
+   * @param day day integer
+   * @return  Calendar instant
+   */
+  @SimpleFunction(description = "Allows the user to set the clock to be "
+    +"a date value.\n" +
+    "Valid values for the month field are 1-12 and 1-31 for the day field.\n")
+  public Calendar MakeDate(int year, int month, int day) {
+    int jMonth = month - 1;
+    try {
+      GregorianCalendar cal = new GregorianCalendar(year, jMonth, day);
+      cal.setLenient(false);
+      cal.getTime();
+    } catch (java.lang.IllegalArgumentException e) {
+      form.dispatchErrorOccurredEvent(this, "ClockMakeDate", ErrorMessages.ERROR_ILLEGAL_DATE);
+    }
+    
+    Calendar instant = Dates.DateInstant(year, month, day);
+    return instant;
+  }
+
+  /**
+   * An instant in time specified by hh:mm
+   * where hh is the hour, mm the minute
+   * @param hour hour integer
+   * @param minute minute integer
+   * @return  Calendar instant since 1/1/1970
+   */
+  @SimpleFunction(description = "Allows the user to set the time of the clock - " +
+    "Valid format is hh:mm\n")
+  public Calendar MakeTime(int hour, int minute) {
+    Calendar instant = null;
+    try {
+      String time_str = String.valueOf(hour) + ":" + String.valueOf(minute);
+      DateFormat timeFormat = new SimpleDateFormat("hh:mm");
+      timeFormat.setLenient(false);
+      Date time = timeFormat.parse(time_str);
+      instant = Calendar.getInstance();
+      instant.setTime(time);
+    }catch (Exception e) {
+    throw new YailRuntimeError(
```

Comment:
Adjust indentation. Also, what is the motivation to throw a `YailRuntimeError` here rather than dispatch an error to the Form like you do in MakeDate?

Reply:
I think it was due to my implementation of parsing the time string. It will give error of ParseException. I will change the implementation to setting hour and minute of Calendar object instead.